src/features/CreationForm/ui/CreationForm/CreationForm.tsx

```diff
@@ -0,0 +1,157 @@
+import { Flex, Modal } from '@mantine/core';
+import { useForm, yupResolver } from '@mantine/form';
+import { ReactNode, useCallback, useMemo, useState } from 'react';
+import { api } from 'shared/api';
+import { Gender } from 'shared/types';
+import {
+    FirstStepValues,
+    FormValues,
+    SecondStepValues,
+    Steps,
+    ThirdStepValues,
+    firstStepSchema,
+    secondStepSchema,
+    thirdStepSchema,
+} from '../../model/types/steps';
+import { CreationStepper } from '../CreationStepper/CreationStepper';
+import { FirstStep } from '../FirstStep/FirstStep';
+import { Message } from '../Message/Message';
+import { SecondStep } from '../SecondStep/SecondStep';
+import { ThirdStep } from '../ThirdStep/ThirdStep';
+
+export interface StepProps {
+    onBack?: () => void;
+    onNext?: () => void;
+    form: ReturnType<typeof useForm<FormValues>>;
+}
+const steps = [Steps.First, Steps.Second, Steps.Third];
+
+export const CreationForm = () => {
+    const [step, setStep] = useState<Steps>(Steps.First);
+    const [message, setMessage] = useState<ReactNode>(null);
+
+    const validate = useCallback(
+        (values: FirstStepValues | SecondStepValues | ThirdStepValues) => {
+            if (step === Steps.First) {
+                return yupResolver(firstStepSchema)(values);
+            }
+            if (step === Steps.Second) {
+                return yupResolver(secondStepSchema)(values);
+            }
+            if (step === Steps.Third) {
+                return yupResolver(thirdStepSchema)(values);
+            }
+            return {};
+        },
+        [step],
+    );
+
+    const onCloseModal = useCallback(() => {
+        setMessage(null);
+    }, [setMessage]);
+
+    const getMessage = useCallback(
+        (result: string | null) => {
+            if (result === null) return null;
+            const title =
+                result === 'success' ? (
+                    <Flex align="center" justify="flex-start">
+                        <b>Форма успешно отправлена</b>
+                    </Flex>
+                ) : (
+                    <Flex align="center" justify="flex-start">
+                        <b>Ошибка</b>
+                    </Flex>
+                );
+            return (
+                <Modal opened centered title={title} onClose={onCloseModal}>
+                    <Message result="success" onClose={onCloseModal} />
+                </Modal>
+            );
+        },
```

Comment:
try it and make it easier, use redux for the `result` and put the markup in the markup